test/model/internal_link_test.dart

```diff
@@ -36,12 +36,17 @@ void main() {
     assert(streams != null || users != null);
     for (final testCase in testCases) {
       final String urlString = testCase.$1;
-      final Narrow? expected = testCase.$2;
-      test(urlString, () async {
-        final store = await setupStore(realmUrl: realmUrl, streams: streams, users: users);
-        final url = store.tryResolveUrl(urlString)!;
-        check(parseInternalLink(url, store)).equals(expected);
-      });
+      // This is here to avoid duplicating each test case that contains `#narrow/stream` or `#narrow/channel`
+      final urlWithChannelSyntax = urlString.replaceFirst('#narrow/stream', '#narrow/channel');
+      final urlWithStreamSyntax = urlString.replaceFirst('#narrow/channel', '#narrow/stream');
+      for (final urlString in {urlWithStreamSyntax, urlWithChannelSyntax}) {
```

Comment:
```suggestion
      for (final urlString in [urlWithStreamSyntax, urlWithChannelSyntax]) {
```

`{'hello'}` creates a `Set<String>`, if that's unintentional use `[` `]` instead, which creates `List<String>`

Reply:
No it is actually intentional. Specifically, when the url being tested does not contain `#narrow/stream` nor `#narrow/channel` then both `urlWithStreamSyntax`, `urlWithChannelSyntax` would be the same and using the list would cause unnecessary duplications.

Reply:
that sounds correct. it's obscure, though. can it get a comment explaining how the object is better than an array?

Reply:
Sure, a new revision is there now.